src/v/cloud_storage/remote.cc

```diff
@@ -485,6 +488,71 @@ ss::future<upload_result> remote::upload_controller_snapshot(
       std::nullopt);
 }
 
+ss::future<remote::jl_write_intent_upload_result>
+remote::jl_write_intent_upload(
+  const cloud_storage_clients::bucket_name& bucket,
+  uint64_t content_length,
+  const reset_input_stream& reset_str,
+  retry_chain_node& parent,
+  lazy_abort_source& lazy_abort_source,
+  std::optional<size_t> max_retries) {
+    uuid_t object_id{};
+    object_id.mutable_uuid() = boost::uuids::random_generator()();
+
+    // TODO(nv): https://en.wikipedia.org/wiki/Year_2038_problem
+    // Divide by 2**6 to get 64 second epochs. The maximum epoch
+    // that can be represented is 2**32 * 2**6 = 2**38 seconds which overflows
+    // around year 10686.
+    auto epoch = static_cast<uint64_t>(
+                   std::chrono::duration_cast<std::chrono::seconds>(
+                     ss::lowres_clock::now().time_since_epoch())
+                     .count())
+                 >> 6;
+
+    object_id.mutable_uuid().data[0] = (epoch >> 24) & 0xFF;
+    object_id.mutable_uuid().data[1] = (epoch >> 16) & 0xFF;
+    object_id.mutable_uuid().data[2] = (epoch >> 8) & 0xFF;
+    object_id.mutable_uuid().data[3] = epoch & 0xFF;
+
+    // We can use alphanumeric listing for garbage collection.
+    // To make sure bad clocks don't mess up, before accepting the write intent
```

Comment:
let's figure out how to get rid of dependency on clocks, or at the very least, estimate how we will get rid of them later. i don't think we should need them.